### ISSUE TITLE
Cirrus: Link Checking Fails For GitHub URLs

### ISSUE THREAD

**Issue by sanssecours** [ElektraInitiative/libelektra#3373]
# Problem Description

It seems that the connection attempts of the [link checker](https://github.com/ElektraInitiative/libelektra/blob/master/scripts/link-checker) in the Cirrus build job `🔗 Check` fail for all GitHub links. To debug this problem I removed the `--quiet` option of the `wget` invocations inside the link checker:

https://github.com/ElektraInitiative/libelektra/blob/7098ab6f12a2f8624107731f4d64e6f3b66b881d/scripts/link-checker#L63-L65

Afterwards the output of the command shows that GitHub returns `429` error codes:

```
Connecting to github.com (github.com)|140.82.113.4|:443... Loaded CA certificate '/etc/ssl/certs/ca-certificates.crt'
Resolving github.com (github.com)... 140.82.113.4
Connecting to github.com (github.com)|140.82.113.4|:443... 140.82.114.3
Connecting to github.com (github.com)|140.82.114.3|:443... connected.
connected.
connected.
connected.
connected.
connected.
connected.
connected.
connected.
HTTP request sent, awaiting response... HTTP request sent, awaiting response... HTTP request sent, awaiting response... HTTP request sent, awaiting response... HTTP request sent, awaiting response... HTTP request sent, awaiting response... HTTP request sent, awaiting response... HTTP request sent, awaiting response... HTTP request sent, awaiting response... 429 too many requests
Remote file does not exist -- broken link!!!
```

It think we need to handle these rate limiting problems. I would recommend we just [whitelist](https://github.com/ElektraInitiative/libelektra/blob/master/tests/linkchecker.whitelist) all links that point to GitHub, until we find a better solution.

**Comment by mpranj** [comment]
Thank you for reporting. I agree that we whitelist the links in the meantime. In the long term I think that a regex-whitelist would be a nicer solution (as suggested in #3374).

**Comment by markus2330** [comment]
The regex-whitelist is quite trivial to do, removing the `-F` from the grep arguments in the link-checker should be enough (assuming that none of our links use chars that have special meaning as regex)..

**Comment by mpranj** [comment]
Closed via #3376.
